### PR TITLE
Feature/m2 732 time range

### PR DESCRIPTION
**Pull request by alex-i-e**
This PR will be merged after M2-774

### CODE REVIEW

File: src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { TimePicker } from 'components';
+import theme from 'styles/theme';
+import { StyledFlexTopStart } from 'styles/styledComponents';
+
+import { ItemOptionContainer } from '../ItemOptionContainer';
+import { TimeRangeProps } from './TimeRange.types';
+
+const timePickerSxProps = {
+  width: '30rem',
+};
```
Comment (kdanilenka32) [inline src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx:14]: space

File: src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+export const TimeRange = <T extends FieldValues>({ name, endTime, control }: TimeRangeProps<T>) => {
+  const { t } = useTranslation('app');
+  const { watch, setValue } = useFormContext();
+  const start = watch(name);
+  const end = watch(endTime);
+
+  useEffect(() => {
+    if (!start || !end || start === end || new Date(start).getTime() <= new Date(end).getTime())
```
Comment (kdanilenka32) [inline src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx:22]: you can move this to var

File: src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import { ItemOptionContainer } from '../ItemOptionContainer';
+import { TimeRangeProps } from './TimeRange.types';
+
+const timePickerSxProps = {
+  width: '30rem',
+};
+export const TimeRange = <T extends FieldValues>({ name, endTime, control }: TimeRangeProps<T>) => {
+  const { t } = useTranslation('app');
+  const { watch, setValue } = useFormContext();
```
Comment (kdanilenka32) [inline src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx:17]: you can get control from here

File: src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx (hunk 1 above)
Comment (alex-i-e) [reply src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx:14]: done

File: src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx (hunk 2 above)
Comment (alex-i-e) [reply src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx]: done

File: src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx (hunk 3 above)
Comment (alex-i-e) [reply src/features/Builder/ActivityItems/ItemConfiguration/InputTypeItems/TimeRange/TimeRange.tsx]: done

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by muhurov — APPROVED** [review]

**Pull request merged**